packages/web-components/src/components/ic-navigation-group/ic-navigation-group.tsx

```diff
@@ -137,23 +137,45 @@ export class NavigationGroup {
 
   private sideNavExpandHandler = (event?: CustomEvent): void => {
     this.isSideNavExpanded = event.detail.sideNavExpanded;
+    const linkWrapper = this.el.shadowRoot.querySelector(
+      this.GROUPED_LINKS_WRAPPER_CLASS
+    ) as HTMLElement;
 
-    if (this.isCollapsedLabelVariant === undefined) {
-      this.isCollapsedLabelVariant = !!(
-        event.target as HTMLIcSideNavigationElement
-      ).getAttribute("collapsed-icon-labels");
-    }
+    if (!linkWrapper) return;
 
-    // Store sum of heights on all collapsed icon label items
-    if (this.isCollapsedLabelVariant && !this.isSideNavExpanded) {
-      this.collapsedIconLabelsNavigationItemsHeight =
-        this.getNavigationChildItemsHeight();
+    if (this.isSideNavExpanded) {
+      if (this.expanded && this.expandedNavItemsHeight) {
+        this.setGroupedLinksElementHeight(
+          linkWrapper,
+          this.expandedNavItemsHeight
+        );
+      } else if (this.expanded) {
+        setTimeout(() => {
+          this.expandedNavItemsHeight = this.getNavigationChildItemsHeight();
+
+          this.setGroupedLinksElementHeight(
+            linkWrapper,
+            this.expandedNavItemsHeight
+          );
+        }, 125);
```

Comment:
Could DYNAMIC_GROUPED_LINKS_HEIGHT_MS be used instead (100ms vs 125ms)?? If not can we add new variable & also use on line 176?